### ISSUE TITLE
No Mesh Pattern Support

### ISSUE THREAD

**Issue by breannansmith** [SimonSapin/cairocffi#81]
cairocffi supports linear and radial patterns, but not mesh patterns. Mesh patterns are an incredibly useful feature of Cairo (for Gourad shading, for example), and would be great to have access to from Python.

**Comment by SimonSapin** [comment]
Yes, this would be nice to have. Cairo’s corresponding C functions can already be used through CFFI with e.g. `cairocffi.cairo.cairo_pattern_create_mesh()`. What’s missing is a new subclass of `cairocffi.Pattern` and its methods.

However I’m not actively working on cairocffi at the moment. I’d review a PR, though.
